middleman-core/lib/middleman-core/dns_resolver.rb

```diff
@@ -0,0 +1,91 @@
+require 'resolv'
+require 'active_support/core_ext/object/blank'
```

Comment:
This one is only for name resolution. The main things here:

* Local Resolver is used from ruby-core -> /etc/hosts on linux and some other path on windows
* this uses HOSTSRC variable to find a local hosts file -> introduced for local testing, but might worth to mention in the docs
* Local Resolver is used first -> then Network Resolver, the network resolver is only used if the local resolver fails.